### ISSUE TITLE
Monitoring service for Registered users, is not synced with the active registered users at the DB.

### ISSUE THREAD

**Issue by leftyb** [RestComm/Restcomm-Connect#1545]
Logs:
https://app.box.com/files/0/f/11998906933/RegisteredUsersnotSyncedmonitoringService

**Comment by gvagenas** [comment]
@leftyb can you please check that LB will always route REGISTRATION of a client to the same node?

**Comment by deruelle** [comment]
@gvagenas @leftyb it may happen that during a failover of the LB, the affinity is lost and the REGISTER affinity for a client would go to a different node

**Comment by gvagenas** [comment]
@deruelle makes sense, I will add patch for UserAgentManager to send updates to MonitoringService so it will sync with the registration updates

**Comment by gvagenas** [comment]
@deruelle @leftyb I have a patch for the issue here but I have some concerns about performance.

The UserAgentManager.clean() method will have to go over all the Registrations in the DB and for those Registrations where the instanceId doesn't match the instanceId of the node, will ask MonitoringService to be removed. This way if a Registration moved to another node and the instanceId changed, UserAgentManager will ask MonitoringService to remove registration.

Problem is that the clean() method will have to go over all registrations.

wdyt?

**Comment by deruelle** [comment]
Bad, let's avoid that, as the userbase will grow it will kill performance completely both DB wise and memory/CPU wise. 

We fetch the current registrations of the instanceid from DB already to send OPTIONS, we could check in memory which one didn't match any of the registrations returned from DB and clean them up, wouldn't it be more effective ?
